pkg/kubectl/cmd/drain.go

```diff
@@ -706,7 +744,12 @@ func (o *DrainOptions) RunCordonOrUncordon(desired bool) error {
 			}
 			unsched := node.Spec.Unschedulable
 			if unsched == desired {
-				cmdutil.PrintSuccess(false, o.Out, nodeInfo.Object, o.DryRun, already(desired))
+				printer, err := o.ToPrinter(already(desired))
```

Comment:
Can we construct appropriate printer once based on different set of input parameters and print once as well, instead of having 3 calls? Btw. each is differently reporting error, which is wrong.

Reply:
I am not sure I understand, do you mean passing the `operation` message to the printer as part of its `PrintObj` method?

Reply:
I mean move this entire block down:
```go
printer, err := o.ToPrinter(<here goes the only thing that changes>)
if err != nil {
    fmt.Fprintf(o.ErrOut, "%v", err)
    continue
}
printer.PrintObj(nodeInfo.AsVersioned(), o.Out)
```
So that we invoke this only once and the only parameter is what you pass to `o.ToPrinter` method.